test/kv_integration_test.go

```diff
@@ -66,25 +67,37 @@ func (kvSuite *KvTestSuite) SetupSuite() {
 	}
 }
 
-func (kvSuite *KvTestSuite) generateRandomSampleFrameWithEmptyStringIndex(size int, indexName string, columnNames []string) frames.Frame {
-	var icol frames.Column
+func (kvSuite *KvTestSuite) generateRandomSampleFrameWithEmptyIndices(size int, indexNames []string, emptyIndices map[string]bool, columnNames []string) frames.Frame {
+	kvSuite.Require().True(len(indexNames) <= 2, "KV API supports no more than two indices")
+	var indexColumns []frames.Column
 
-	index := make([]string, size)
+	emptyIndexValues := make([]string, size)
+	uniqueIndexValues := make([]string, size)
 	for i := 0; i < size; i++ {
-		index[i] = ""
+		emptyIndexValues[i] = fmt.Sprintf("%d", i)
```

Comment:
isn't this the `uniqueIndexValues`?

Reply:
also this means that the errors you checked for are not right.. maybe add a more specific  error check then `"should not be empty"`